dot/node.go

```diff
@@ -171,28 +215,32 @@ func LoadGlobalNodeName(basepath string) (nodename string, err error) {
 	nodename, err = basestate.LoadNodeGlobalName()
 	if err != nil {
 		logger.Warnf("failed to load global node name from base path %s: %s", basepath, err)
-		return "", err
 	}
-
 	return nodename, err
 }
 
-// NewNode creates a new dot node from a dot node configuration
+// NewNode to create node based on given Config
 func NewNode(cfg *Config, ks *keystore.GlobalKeystore) (*Node, error) {
+	return newNode(cfg, ks, nodeBuilder{})
+}
+
+func newNode(cfg *Config, ks *keystore.GlobalKeystore, nbi nodeBuilderIface) (*Node, error) {
```

Comment:
nit rename `nbi` to `builder` or `nodeBuilder`

Reply:
Updated to builder.